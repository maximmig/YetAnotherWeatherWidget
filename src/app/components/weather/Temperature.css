.temp {
    font-size: 1em;
    line-height: 1;
}

.tempUnit {
    font-size: .5em;
    vertical-align: super;
}