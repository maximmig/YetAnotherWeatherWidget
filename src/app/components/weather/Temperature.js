import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import {UNITS} from '../../constants';

import styles from './Temperature.css';

const Temperature = ({selectedUnits, temp, className}) => {
  const classes = classnames(styles.temp, className);
  const roundedTemp = Math.round(temp);
  const tempUnit = UNITS[selectedUnits].temp;

  return (
    <span className={classes}>
      {roundedTemp}<span className={styles.tempUnit}>{tempUnit}</span>
    </span>
  );
};

Temperature.propTypes = {
  selectedUnits: PropTypes.oneOf(Object.keys(UNITS)).isRequired,
  temp: PropTypes.number.isRequired,
  className: PropTypes.string,
};

export default Temperature;
