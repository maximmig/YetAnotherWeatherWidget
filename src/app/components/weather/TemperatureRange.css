:root {
  --temperature-margin: 5px;
}

.tempRange {
  font-size: 3em;
  line-height: 1;
}

.tempEdge {
  opacity: .6;
  font-size: .6em;
}

.minTemp {
  color: #008;
  margin-right: var(--temperature-margin);
}

.minTemp::before {
  color: currentColor;
  content: '↓';
}

.maxTemp {
  color: #800;
  margin-left: var(--temperature-margin);
}

.maxTemp::after {
  color: currentColor;
  content: '↑';
}
