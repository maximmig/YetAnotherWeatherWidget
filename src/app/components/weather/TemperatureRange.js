import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import {UNITS} from '../../constants';

import Temperature from './Temperature';

import styles from './TemperatureRange.css';

const TemperatureRange = ({selectedUnits, temp, minTemp, maxTemp}) => {
  const minMaxTempsDontMatch = minTemp !== maxTemp;
  return (
    <div className={styles.tempRange}>
      {minMaxTempsDontMatch && (
        <Temperature
          className={classnames(styles.tempEdge, styles.minTemp)}
          selectedUnits={selectedUnits}
          temp={minTemp}
        />
      )}
      <Temperature
        selectedUnits={selectedUnits}
        temp={temp}
      />
      {minMaxTempsDontMatch && (
        <Temperature
          className={classnames(styles.tempEdge, styles.maxTemp)}
          selectedUnits={selectedUnits}
          temp={maxTemp}
        />
      )}
    </div>
  );
};

TemperatureRange.propTypes = {
  selectedUnits: PropTypes.oneOf(Object.keys(UNITS)).isRequired,
  temp: PropTypes.number.isRequired,
  minTemp: PropTypes.number,
  maxTemp: PropTypes.number,
};

export default TemperatureRange;
