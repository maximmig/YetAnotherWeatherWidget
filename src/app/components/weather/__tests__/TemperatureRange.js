import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TemperatureRange from '../TemperatureRange';
import Temperature from '../Temperature';

import styles from '../TemperatureRange.css';

Enzyme.configure({adapter: new Adapter()});

describe('TemperatureRange component', () => {
  const selectedUnits = 'metric';
  const minTemp = 10;
  const temp = 20;
  const maxTemp = 30;

  it('should render a Temperature component for each temperature', () => {
    const wrapper = shallow(
      <TemperatureRange
        selectedUnits={selectedUnits}
        temp={temp}
        minTemp={minTemp}
        maxTemp={maxTemp}
      />
    );
    expect(wrapper.find(Temperature).length).toEqual(3);
    expect(wrapper.find(`.${styles.minTemp}`).length).toEqual(1);
    expect(wrapper.find(`.${styles.maxTemp}`).length).toEqual(1);
  });

  it('should not render Temperature components for min and max temperatures if they are equal', () => {
    const wrapper = shallow(
      <TemperatureRange
        selectedUnits={selectedUnits}
        temp={temp}
        minTemp={temp}
        maxTemp={temp}
      />
    );
    expect(wrapper.find(Temperature).length).toEqual(1);
    expect(wrapper.find(`.${styles.minTemp}`).length).toEqual(0);
    expect(wrapper.find(`.${styles.maxTemp}`).length).toEqual(0);
  });

  it('should not render Temperature components for min and max temperatures if omitted', () => {
    const wrapper = shallow(
      <TemperatureRange
        selectedUnits={selectedUnits}
        temp={temp}
      />
    );
    expect(wrapper.find(Temperature).length).toEqual(1);
    expect(wrapper.find(`.${styles.minTemp}`).length).toEqual(0);
    expect(wrapper.find(`.${styles.maxTemp}`).length).toEqual(0);
  });

  it('should match the last snapshot with average temperature only', () => {
    const wrapper = shallow(
      <TemperatureRange
        selectedUnits={selectedUnits}
        temp={temp}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match the last snapshot with min, max and average temperatures', () => {
    const wrapper = shallow(
      <TemperatureRange
        selectedUnits={selectedUnits}
        temp={temp}
        minTemp={minTemp}
        maxTemp={maxTemp}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
