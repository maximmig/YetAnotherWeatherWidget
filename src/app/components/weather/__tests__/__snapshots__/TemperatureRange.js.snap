// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureRange component should match the last snapshot with average temperature only 1`] = `
<div
  className="tempRange"
>
  <Temperature
    selectedUnits="metric"
    temp={20}
  />
</div>
`;

exports[`TemperatureRange component should match the last snapshot with min, max and average temperatures 1`] = `
<div
  className="tempRange"
>
  <Temperature
    className="tempEdge minTemp"
    selectedUnits="metric"
    temp={10}
  />
  <Temperature
    selectedUnits="metric"
    temp={20}
  />
  <Temperature
    className="tempEdge maxTemp"
    selectedUnits="metric"
    temp={30}
  />
</div>
`;
